mods/page_revisions.js

```diff
@@ -253,8 +253,9 @@ PRS.prototype.getTitleRevision = function(restbase, req) {
         throw new Error("Invalid revision: " + rp.revision);
     }
     return revisionRequest
-    .then(self._checkRevReturn)
     .then(function(res) {
+        // check if the revision has any restrictions
+        self._checkRevReturn(res);
```

Comment:
There is another one at https://github.com/wikimedia/restbase/blob/phab/T76165/mods/page_revisions.js#L343